torch/distributed/_functional_collectives.py

```diff
@@ -291,6 +291,32 @@ def reduce_scatter_tensor_coalesced(
     return list(map(_maybe_wrap_tensor, tensor_list))
 
 
+def all_to_all_single(
+    self: torch.Tensor,
+    output_split_sizes: List[int],
```

Comment:
We cannot use object list for input/output split sizes, they must be tensors or dynamo will incorrectly bake them as constants.

Reply:
I was thinking that we should probably follow the xla semantic of alltoall https://www.tensorflow.org/xla/operation_semantics#alltoall, that would make dynamo and inductor be much happier. The output_split_sizes and input_split_sizes would require pass in constant list (even if we make them as tensor, inductor technically can't do anything better rather than be blind on it).